holidays/countries/azerbaijan.py

```diff
@@ -11,195 +11,214 @@
 
 from datetime import date
 from datetime import timedelta as td
-from typing import Dict, List
 
-from holidays.calendars import _islamic_to_gre
 from holidays.constants import JAN, MAR, APR, MAY, JUN, JUL, AUG, SEP, OCT
 from holidays.constants import NOV, DEC
 from holidays.holiday_base import HolidayBase
+from holidays.holiday_groups import InternationalHolidays, IslamicHolidays
 
 
-class Azerbaijan(HolidayBase):
+class Azerbaijan(HolidayBase, InternationalHolidays, IslamicHolidays):
     # [1] https://en.wikipedia.org/wiki/Public_holidays_in_Azerbaijan
     # [2] https://az.wikipedia.org/wiki/Az%C9%99rbaycan%C4%B1n_d%C3%B6vl%C9%99t_bayramlar%C4%B1_v%C9%99_x%C3%BCsusi_g%C3%BCnl%C9%99ri  # noqa: E501
     # [3] https://www.sosial.gov.az/en/prod-calendar
 
     country = "AZ"
 
-    def _populate(self, year: int) -> None:
-        def collect_holiday(hol_date: date, hol_name: str) -> None:
-            if hol_date in holiday_date_names_mapping:
-                holiday_date_names_mapping[hol_date].append(hol_name)
+    def __init__(self, *args, **kwargs):
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(self)
+        super().__init__(*args, **kwargs)
+
+    def _populate(self, year):
+        def _add_observed(dt: date, name: str = None):
```

Comment:
For some reason I thought we agreed to have `_add_[with]_obervered` related methods unnested.

I guess we'll refactor that later. Looks good to me for now.

Reply:
Here `_add_observed()` uses a local variable `hol_dates`, so I left it internal function 🙄